ai/src/train.py

```diff
@@ -107,18 +139,30 @@ def train(self):
                 episode_start = math.floor(time.time())
                 self.data_manager.reset()
                 raw_state = self.data_manager.get_current_window()
-                raw_state_prime_interpretations = self.data_manager.get_interpretations_for_interval()
+                raw_state_prime_interpretations = (
+                    self.data_manager.get_interpretations_for_interval()
+                )
                 model_state = self.data_manager.flatten_and_normalize_window(raw_state)
 
                 total_steps = (
-                    math.floor(self.data_manager.param.period_secs / self.data_manager.param.granularity_secs)
-                    if isinstance(self.data_manager, TimeSeriesDataManager) else
-                    len(self.data_manager.data_frame))
-                progress_bar = ProgressBar(self.pod_name, episode, total_steps, self.data_manager.metrics)
+                    math.floor(
```

Comment:
integer division can be used here: `self.data_manager.param.period_secs // self.data_manager.param.granularity_secs`